Framework/MDAlgorithms/src/AccumulateMD.cpp

```diff
@@ -0,0 +1,467 @@
+#include "MantidMDAlgorithms/AccumulateMD.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/ArrayBoundedValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/ListValidator.h"
+#include "MantidKernel/MandatoryValidator.h"
+#include "MantidKernel/PropertyWithValue.h"
+#include "MantidAPI/FileFinder.h"
+#include "MantidAPI/HistoryView.h"
+#include "MantidDataObjects/MDHistoWorkspaceIterator.h"
+#include <Poco/File.h>
+
+using namespace Mantid::Kernel;
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+
+namespace Mantid {
+namespace MDAlgorithms {
+
+/*
+ * Reduce the vector of input data to only data files and workspaces which can
+ * be found
+ * @param input_data :: Vector of input data sources
+ * @param psi :: Vector of goniometer angle psi containing a value for each data
+ * source
+ * @param gl :: Vector of goniometer anglegl containing a value for each data
+ * source
+ * @param gs :: Vector of goniometer angle gs containing a value for each data
+ * source
+ * @param efix :: Vector of data source energy values in meV
+*/
+void filterToExistingSources(std::vector<std::string> &input_data,
+                             std::vector<double> &psi, std::vector<double> &gl,
+                             std::vector<double> &gs,
+                             std::vector<double> &efix) {
+  for (size_t i = input_data.size(); i > 0; i--) {
+    if (!dataExists(input_data[i - 1])) {
```

Comment:
Does this silently ignore entries that can't be found? Would we want to log, or better yet, return anything that couldn't currently be found?